sdc/tests/tests_perf/test_perf_df_rolling.py

```diff
@@ -0,0 +1,136 @@
+        rolling_params.append(f'min_periods={min_periods}')
+
+    return ', '.join(rolling_params)
+
+
+def gen_df_rolling_usecase(method_name, rolling_params=None, method_params=''):
+    """Generate df rolling method use case"""
+    if not rolling_params:
+        rolling_params = get_rolling_params()
+
+    func_text = rolling_usecase_tmpl.format(**{
+        'method_name': method_name,
+        'rolling_params': rolling_params,
+        'method_params': method_params
+    })
+
+    global_vars = {'time': time}
+    loc_vars = {}
+    exec(func_text, global_vars, loc_vars)
+    _df_rolling_usecase = loc_vars[f'df_rolling_{method_name}_usecase']
+
+    return _df_rolling_usecase
+
+
+# python -m sdc.runtests sdc.tests.tests_perf.test_perf_df_rolling.TestDFRollingMethods
+class TestDFRollingMethods(TestBase):
+    # more than 19 columns raise SystemError: CPUDispatcher() returned a result with an error set
+    max_columns_num = 19
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.total_data_length = {
+            'min': [2 * 10 ** 5],
+        }
+
+    def _test_jitted(self, pyfunc, record, *args, **kwargs):
+        # compilation time
+        record['compile_results'] = calc_compilation(pyfunc, *args, **kwargs)
+
+        cfunc = numba.njit(pyfunc)
+
+        # Warming up
+        cfunc(*args, **kwargs)
+
+        # execution and boxing time
+        record['test_results'], record['boxing_results'] = get_times(cfunc, *args, **kwargs)
+
+    def _test_python(self, pyfunc, record, *args, **kwargs):
+        record['test_results'], _ = get_times(pyfunc, *args, **kwargs)
+
+    def _test_case(self, pyfunc, data_name, test_name=None,
+                   input_data=test_global_input_data_float64, columns_num=10):
+        if columns_num > self.max_columns_num:
+            columns_num = self.max_columns_num
+
+        test_name = test_name or data_name
+
+        full_input_data_length = sum(len(i) for i in input_data)
+        for data_length in self.total_data_length[data_name]:
+            base = {
+                'test_name': test_name,
+                'data_size': data_length,
+            }
+            data = perf_data_gen_fixed_len(input_data, full_input_data_length, data_length)
+            test_data = pandas.DataFrame({col: data for col in string.ascii_uppercase[:columns_num]})
+
+            record = base.copy()
+            record['test_type'] = 'SDC'
+            self._test_jitted(pyfunc, record, test_data)
+            self.test_results.add(**record)
+
+            record = base.copy()
+            record['test_type'] = 'Python'
+            self._test_python(pyfunc, record, test_data)
+            self.test_results.add(**record)
+
+    def _test_df_rolling_method(self, name):
+        usecase = gen_df_rolling_usecase(name)
+        test_name = f'DataFrame.rolling.{name}'
+        self._test_case(usecase, name, test_name=test_name)
+
+    def test_df_rolling_min(self):
```

Comment:
This is function can be obtained using template for tests. For example, how in test_perf_series.py

Reply:
I tried to implement such generator:
```
test_params = {
    'min': {'rolling': None, 'method': ''}
}

for name, params in test_params.items():
    test_name = f'test_df_rolling_{name}'
    test_text = f"""
    def {test_name}(self):
        self._test_df_rolling_method("{name}", rolling_params={params['rolling']}, method_params={params['method']})
    """

    loc_vars = {}
    exec(test_text, {}, loc_vars)
    _df_rolling_test = loc_vars[test_name]

    setattr(TestDFRollingMethods, test_name, _df_rolling_test)
```

My opinion the generator is excess because actually there are no real advantages from that, e.g. we we don't even shorten the code.

Reply:
if add in test_params in values "data_length" then will not need  to conect with list "total_data_length" in class. Also definition parameters will in one place that convenient.

Reply:
I agree with that, but I don't like how the generator looks (it looks harder than tests itself) and I don't see real advantages of that.

Reply:
@Rubtsowa in conclusion, I propose to implement generic test generator in separate PR which can be used for generating most the performance tests. In this case we could use it here as well.